### PR TITLE
Add token field in ticket

### PR DESCRIPTION
**Pull request by siddhantbajaj**
- [x] Token field in physical ticket

### CODE REVIEW

**Comment by siddhantbajaj** [comment]
@Ana06 
Sorry I didn’t knew those task were meant to be done in order.
I though it would be better to have the Qr that need to scanned ready first and then jump to the scanner.

**Comment by Ana06** [comment]
@siddhantbajaj 

> @Ana06
> Sorry I didn’t knew those task were meant to be done in order.
> I though it would be better to have the Qr that need to scanned ready first and then jump to the scanner because in that way it would easier to define the routes and all the scanning logic . I suppose we need to have to the token field for reading of QR.

Then you will render a QR that is not working. Let's do it the other way around and then we can reopen this again :wink:

File: app/models/physical_ticket.rb [hunk 1]
```diff
@@ -4,4 +4,19 @@ class PhysicalTicket < ActiveRecord::Base
...
+  private
+
+  def set_token
+    self.token = generate_token
+  end
+
+  def generate_token
+    loop do
+      token = SecureRandom.hex(10)
+      break token unless PhysicalTicket.where(token: token).exists?
```
Comment (Ana06) [inline app/models/physical_ticket.rb]: I think you can also write this like:

`break token unless PhysicalTicket.exists?(token: token)`

Comment (siddhantbajaj) [inline app/models/physical_ticket.rb]: 👍

**Pull request merged**